esm_runscripts/compute.py

```diff
@@ -505,16 +508,31 @@ def copy_tools_to_thisrun(config):
 
     # check for recursive creation of the file tree. This prevents the risk of
     # creating a run directory tree inside the `scriptsdir`
-    scriptsdir_inside_fromdir = pathlib.Path(fromdir) in pathlib.Path(scriptsdir).parents
-    if scriptsdir_inside_fromdir and config["general"]["verbose"]:
-        print("WARNING: scriptsdir is inside fromdir")
-        print(f"  - scriptsdir: {scriptsdir}")
-        print(f"  - fromdir:    {fromdir}")
+    # example:
+    # fromdir:    /mnt/lustre02/work/project/username/basedir
+    # scriptsdir: /mnt/lustre02/work/project/username/basedir/expid/scripts
+    # it is assumed that if 5th parent of the scriptsdir is inside the fromdir
+    # then probably some sort of infinite recursion is entered. These lines
+    # protect such problems
+    scriptsdir_deep_parents = list(pathlib.Path(scriptsdir).parents)[5:]
```

Comment:
I don't really like the hardcoded 5 here....this is **very** specific to how mistral's filesystem is set up.

Reply:
5 is more like a threshold for detecting infinite recursion. Typically `esm_tools` directory tree is not very deep. This is not really about mistral since we start counting 5 from the deepest level (right to left). It is machine agnostic.